### ISSUE TITLE
Path to Haxe executable is not valid: '${workspaceFolder}/node_modules/.bin/haxe'. 

### ISSUE THREAD

**Issue by darmie** [vshaxe/vshaxe#503]
I am using lix, and I am trying to force vshaxe to use the Haxe executable that is within the project scope... But I am getting this error 

```
Path to Haxe executable is not valid: '${workspaceFolder}/node_modules/.bin/haxe'. Please check your settings.
```

This is what I set in Settings.json

```json
"haxe.executable": {
        "path": "${workspaceFolder}/node_modules/.bin/haxe",
        "env": {
            "PATH": "${workspaceFolder}/node_modules/.bin/haxelib"
        }
    }
```

**Comment by Gama11** [comment]
> I am using lix, and I am trying to force vshaxe to use the Haxe executable that is within the project scope...

Just install the lix vscode extension? :)

**Comment by darmie** [comment]
@Gama11 I already have Lix installed, and I am always getting `Error: env: node: No such file or directory` That is why I am trying to set it myself

**Comment by AlexHaxe** [comment]
do you have a node binary in your path? or what happens when you do `/usr/bin/env node` (from VSCode terminal and from a dedicated command shell)?

**Comment by darmie** [comment]
@AlexHaxe I do!

![Screenshot 2021-08-23 at 13 13 38](https://user-images.githubusercontent.com/848002/130445150-623318c8-e84b-4582-a3a4-b3cc30b28967.png)

**Comment by Gama11** [comment]
Is that the VSCode terminal?

**Comment by darmie** [comment]
Yes it is, I am using zsh on Mac.

**Comment by AlexHaxe** [comment]
does lix / Haxe work in a normal shell (outside of VSCode)?
and what's the output of running `npx haxe` (in and outside of VSCode) and `~/haxe/versions/4.2.3/haxe`?

**Comment by darmie** [comment]
All the commands work in and out of vscode terminal.

It seems like it is either the vshaxe or Lix extension that's unable to run
the commands.

**Comment by darmie** [comment]
I should also note that I updated my vscode recently.

**Comment by Gama11** [comment]
>I should also note that I updated my vscode recentl

And did it used to work before you did?

**Comment by darmie** [comment]
Yes it used to work before.

**Comment by Gama11** [comment]
Neither vshaxe nor the lix extension changed recently, so it very well could be related to the VSCode update.

**Comment by back2dos** [comment]
Perhaps they've stopped to automatically interpolate strings settings (so that extensions may do this at their own discretion)?

**Comment by Gama11** [comment]
I don't think they ever did that.

**Comment by back2dos** [comment]
How else would the above config have ever worked then? ^^

**Comment by darmie** [comment]
even after I removed the config, and allow the extensions to do their work, I always get this `Error: env: node: No such file or directory`

**Comment by darmie** [comment]
I still have this problem till date. I struggle to use lix

**Comment by back2dos** [comment]
What error do you get without *any* setting but with the lix extension installed?

**Comment by darmie** [comment]
I had to remove the custom settings and restarted the whole vscode app again. Then it worked. 

But anytime I start a new project, it doesn't work out of the box.

**Comment by back2dos** [comment]
> But anytime I start a new project, it doesn't work out of the box.

Hmm, not sure what you mean. If you have the lix extension and no custom settings, then it should work - "out of the box". Are you creating your new projects with a tool that generates this settings.json?

**Comment by darmie** [comment]
No. The settings.json was there because it didn't work out of the box and I tried to fix it

**Comment by darmie** [comment]
When I start a new lix project, the syntax completion won't work, and when I check the logs it would say something about "haxe(:env)" not found.  That is why I attempt to fix it with settings.json

**Comment by back2dos** [comment]
But I thought it only works when you *remove* the setting, or how is this to be understood?

> I had to remove the custom settings and restarted the whole vscode app again. Then it worked.
